dotnet/src/Experimental/Process.LocalRuntime/LocalStep.cs

```diff
@@ -39,47 +40,37 @@ internal class LocalStep : KernelProcessMessageChannel
     /// <summary>
     /// Represents a step in a process that is running in-process.
     /// </summary>
-    /// <param name="name">Required. The name of the step.</param>
-    /// <param name="id">Required. The unique Id of the step.</param>
+    /// <param name="stepInfo">An instance of <see cref="KernelProcessStepInfo"/></param>
     /// <param name="kernel">Required. An instance of <see cref="Kernel"/>.</param>
     /// <param name="parentProcessId">Optional. The Id of the parent process if one exists.</param>
     /// <param name="loggerFactory">An instance of <see cref="LoggerFactory"/> used to create loggers.</param>
-    public LocalStep(string name, string id, Kernel kernel, string? parentProcessId = null, ILoggerFactory? loggerFactory = null)
+    public LocalStep(KernelProcessStepInfo stepInfo, Kernel kernel, string? parentProcessId = null, ILoggerFactory? loggerFactory = null)
     {
-        Verify.NotNullOrWhiteSpace(name);
-        Verify.NotNullOrWhiteSpace(id);
+        Verify.NotNull(stepInfo);
         Verify.NotNull(kernel);
+        Verify.NotNullOrWhiteSpace(stepInfo.State.Name);
 
-        this.Name = name;
-        this.Id = id;
+        this.Id = stepInfo.State.Id ?? Guid.NewGuid().ToString("n");
+        this.Name = stepInfo.State.Name;
         this.ParentProcessId = parentProcessId;
         this.LoggerFactory = loggerFactory;
         this._kernel = kernel;
+        this._stepInfo = stepInfo;
+        this._initializeTask = new Lazy<ValueTask>(this.InitializeStepAsync);
+        this._logger = this.LoggerFactory?.CreateLogger(this._stepInfo.InnerStepType) ?? new NullLogger<LocalStep>();
+        this._outputEdges = this._stepInfo.Edges.ToDictionary(kvp => kvp.Key, kvp => kvp.Value.ToList());
     }
 
     /// <summary>
     /// Initializes the step with the provided step information.
     /// </summary>
-    /// <param name="stepInfo">An instance of <see cref="KernelProcessStepInfo"/></param>
     /// <returns>A <see cref="ValueTask"/></returns>
     /// <exception cref="KernelException"></exception>
-    public virtual async ValueTask InitializeAsync(KernelProcessStepInfo stepInfo)
+    internal virtual async ValueTask InitializeStepAsync()
```

Comment:
Can this be made private if initialization should happen via `this._initializeTask`?